Server/Controllers/users.js

```diff
@@ -0,0 +1,368 @@
+import lodash from 'lodash';
+import groupController from './groups';
+import emailValidator from '../Utilities/emailValidator';
+
+export default {
+/**
+   * Register New User
+   * Route: POST: /users/signup
+   * @param {Object} req request object
+   * @param {Object} res response object
+   * @param {firebase} firebase firebase object
+   * @returns {response} Response object
+   */
+
+  signUp(req, res, firebase) {
+    const { email, password, userName, photoURL, phoneNo } = req.body;
+    if (!emailValidator(email)) {
+      res.status(401).send({
```

Comment:
Should this be a 401 or a 400? Do well to take  a good look at status codes and their use cases, and make appropriate changes.